amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/proxy/v2/AmqpProxyClientChannel.java

```diff
@@ -0,0 +1,220 @@
+ */
+package io.streamnative.pulsar.handlers.amqp.proxy.v2;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.qpid.server.bytebuffer.QpidByteBuffer;
+import org.apache.qpid.server.protocol.v0_8.AMQShortString;
+import org.apache.qpid.server.protocol.v0_8.FieldTable;
+import org.apache.qpid.server.protocol.v0_8.transport.BasicConsumeOkBody;
+import org.apache.qpid.server.protocol.v0_8.transport.BasicContentHeaderProperties;
+import org.apache.qpid.server.protocol.v0_8.transport.ChannelCloseOkBody;
+import org.apache.qpid.server.protocol.v0_8.transport.ClientChannelMethodProcessor;
+import org.apache.qpid.server.protocol.v0_8.transport.ExchangeDeclareOkBody;
+import org.apache.qpid.server.protocol.v0_8.transport.QueueBindOkBody;
+import org.apache.qpid.server.protocol.v0_8.transport.QueueDeclareOkBody;
+import org.apache.qpid.server.protocol.v0_8.transport.QueueUnbindOkBody;
+
+
+@Slf4j
+public class AmqpProxyClientChannel implements ClientChannelMethodProcessor {
+
+    private final Integer channelId;
+    private final ProxyBrokerConnection conn;
+
+    public AmqpProxyClientChannel(Integer channelId, ProxyBrokerConnection conn) {
+        this.channelId = channelId;
+        this.conn = conn;
+    }
+
+    @Override
+    public void receiveChannelOpenOk() {
+        conn.getChannelState().put(channelId, ProxyBrokerConnection.ChannelState.OPEN);
+        conn.getProxyConnection().getServerChannelMap().get(channelId).initComplete();
+    }
+
+    @Override
+    public void receiveChannelAlert(int replyCode, AMQShortString replyText, FieldTable details) {
+        // nothing to do
+    }
+
+    @Override
+    public void receiveAccessRequestOk(int ticket) {
+        // nothing to do
+    }
+
+    @Override
+    public void receiveExchangeDeclareOk() {
+        conn.getClientChannel().writeAndFlush(new ExchangeDeclareOkBody().generateFrame(channelId));
+    }
+
+    @Override
+    public void receiveExchangeDeleteOk() {
+        // nothing to do
+    }
+
+    @Override
+    public void receiveExchangeBoundOk(int replyCode, AMQShortString replyText) {
+        // nothing to do
+    }
+
+    @Override
+    public void receiveQueueBindOk() {
+        conn.getClientChannel().writeAndFlush(new QueueBindOkBody().generateFrame(channelId));
+    }
+
+    @Override
+    public void receiveQueueUnbindOk() {
+        conn.getClientChannel().writeAndFlush(new QueueUnbindOkBody().generateFrame(channelId));
+    }
+
+    @Override
+    public void receiveQueueDeclareOk(AMQShortString queue, long messageCount, long consumerCount) {
+        conn.getClientChannel().writeAndFlush(
+                new QueueDeclareOkBody(queue, messageCount, consumerCount).generateFrame(channelId));
+    }
+
+    @Override
+    public void receiveQueuePurgeOk(long messageCount) {
+        // nothing to do
+    }
+
+    @Override
+    public void receiveQueueDeleteOk(long messageCount) {
+        // nothing to do
```

Comment:
The client needs to be notified of the result.

Reply:
This method will be supported later.